ddl/reorg.go

```diff
@@ -159,6 +187,14 @@ func (w *worker) runReorgJob(t *meta.Meta, reorgInfo *reorgInfo, tblInfo *model.
 		logutil.BgLogger().Info("[ddl] run reorg job done", zap.Int64("handled rows", rowCount))
 		// Update a job's RowCount.
 		job.SetRowCount(rowCount)
+
+		// Update a job's warnings.
+		w.reorgCtx.mu.Lock()
+		partWarnings := w.reorgCtx.mu.warnings
+		partWarningsCount := w.reorgCtx.mu.warningsCount
+		job.SetWarnings(mergeWarningsAndWarningsCount(partWarnings, job.ReorgMeta.Warnings, partWarningsCount, job.ReorgMeta.WarningsCount))
+		w.reorgCtx.mu.Unlock()
```

Comment:
Could we extract this code into a function? It's duplicated code.

Reply:
good